### PR TITLE
TurboWarp redirect

### PR DESCRIPTION
**Pull request by illogicalapple**
### Changes
Added turbowarp redirect addon

### Reason for changes
Redirects unshared projects to Turbowarp

### Tests
Everything works!
I have tested it by installing a copy of Scratch Addons

### CODE REVIEW

File: addons/turbowarp-redirect/addon.json [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        {
+          "id": "//forkphorus.github.io/#",
+          "name": "Forkphorus"
+        }
+      ]
+    }
+  ],
+  "versionAdded": "1.26.0",
+  "dynamicEnable": true,
+  "enabledByDefault": true
```
Comment (Norbiros) [inline addons/turbowarp-redirect/addon.json:37]: I think it should be `false`.

File: addons/turbowarp-redirect/addon.json [hunk 2]
```diff
@@ -0,0 +1,38 @@
+{
+  "name": "Turbowarp redirect",
```
Comment (CST1229) [inline addons/turbowarp-redirect/addon.json:2]:
```suggestion
  "name": "TurboWarp redirect",
```
TurboWarp is a name.

**Review by Norbiros — CHANGES_REQUESTED** [review]

File: addons/turbowarp-redirect/addon.json [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  "settings": [
+  	{
+      "name": "Redirect to",
+      "id": "redirectUrl",
+      "type": "select",
+      "default": "//turbowarp.org/",
+      "potentialValues": [
+        {
+          "id": "//turbowarp.org/",
+          "name": "Turbowarp"
```
Comment (CST1229) [inline addons/turbowarp-redirect/addon.json:26]:
```suggestion
  "description": "Redirects unshared projects to TurboWarp.",
  "tags": ["community"],
  "credits": [
    {
      "name": "illogicalapple",
      "link": "https://scratch.mit.edu/users/illogicalapple/"
    }
  ],
  "userscripts": [
    {
      "url": "userscript.js",
      "matches": ["https://scratch.mit.edu/projects/*"]
    }
  ],
  "settings": [
  	{
      "name": "Redirect to",
      "id": "redirectUrl",
      "type": "select",
      "default": "//turbowarp.org/",
      "potentialValues": [
        {
          "id": "//turbowarp.org/",
          "name": "TurboWarp"
```
(also in these places)

**Comment by pufferfish101007** [comment]
Pretty sure this has been rejected for ethical reasons.

**Comment by Norbiros** [comment]
> Pretty sure this has been rejected for ethical reasons.

I agree!

Screenshot from 2022-04-09 17-36-29

I think it is in some kind "exploiting bug", scratch wanted to share projects, so you can hide them. They didn't block api, but when we are creating simple way to see unshared projects, it is not nice :-/. But as always, the last word belongs to W_L.

**Comment by mxmou** [comment]
> Pretty sure this has been rejected for ethical reasons.

#510 is very similar to this PR and was rejected. #3222 suggests adding a link to an alternative player instead and is apparently not rejected.

File: addons/turbowarp-redirect/addon.json (hunk 1 above)
Comment (illogicalapple) [reply addons/turbowarp-redirect/addon.json:37]: Yes, I forgot about that, thanks! :)

**Comment by illogicalapple** [comment]
> Pretty sure this has been rejected for ethical reasons.

https://docs.turbowarp.org/unshared-projects
Besides, there's [probably] no way you would be able to get a random project ID of a random unshared project without someone giving it to you.

**Comment by Chiroyce1** [comment]
Maybe some text on screen that says `Redirecting you to TurboWarp...` while it redirects? It will only be shown for <= 1 second(s) but still...

**Comment by Norbiros** [comment]
Screenshot from 2022-04-09 17-33-29
I think that we can display scratch tool bar to, if you want to go back etc...

**Comment by WorldLanguages** [comment]
"See deleted comments" and an addon like this were rejected by griffpatch back in 2020.

**Comment by Norbiros** [comment]
@WorldLanguages so please, close also #3222. If it is rejected  other issues like 3222 should be rejected too

**Comment by WorldLanguages** [comment]
@Norbiros Thanks!

See #3222, I just posted more details about what addons griffpatch rejected in 2020.

**Pull request closed without merging**